content/en/blog/wireshark.md

```diff
@@ -119,63 +123,90 @@ service PersonSearchService {
-end
-```
+Because the service uses the `Person` type defined in `addressbook.proto`,
+the address book `.proto` is imported at the start of the file.
+
+### Setting protobuf search paths
+
+Wireshark gives the most meaningful decodings when it knows about the `.proto`
+files used by the apps whose messages you are analyzing.
+
+You can tell Wireshark where to find `.proto` files by setting the **Protobuf
+Search Paths** in the preferences accessible from the **Edit** menu under
+**Preferences \> Protocols \> Protobuf**.
+
+If our example app's `.proto` files are in the `d:/protos/my_proto_files` directory,
+and the official Protobuf library directory is
+`d:/protos/protobuf-3.4.1/include`, then add these two paths as _source
+directories_ like this:
+
+![Protobuf-search-paths dialog](/img/wireshark_protobuf_search_paths.png)
+
+By selecting the **Load all files** option for the app's protocol directory you
+enable preloading of message definitions from the `addressbook.proto` file.
+
+### Loading a capture file
+
+From the Wireshark [SampleCaptures page][], download the following sample gRPC
+capture file created by running the app and issuing a search request:
+[grpc_person_search_protobuf_with_image.pcapng][].
+
+Load this file in Wireshark by selecting **Open** from the **File** menu.
+
+### Setting port traffic type
+
+The app's gRPC traffic was captured on port 50051. Before you can decode the
+port's messages, you need to register the port as carrying HTTP2 traffic; do
+this through the **Decode As** dialog, which you access from the **Analyze**
+menu:
+
+![Decode-as dialog](/img/wireshark_decode_as_dialog.png)
+
+### Decoding the search request message
+
+<!--
+TODO: What are the steps that the user needs to follow before the user can see
+      the decoded message as shown in the screenshot?
+-->
+Select the first gRPC message sent to port 50051, it corresponds to the sample's
+service request message. This is how Wireshark dissects the gRPC request:
+
+<!--
+TODO: I suggest reverting to the layout of the previous image, which didn't
+      show the bytes pane. Also collapsing the "Stream: DATA" would be a good idea.
+-->
+![Decoded search request](/img/wireshark_grpc_protobuf_search_request.png)
+
+By examining the HTTP2 message header `path` field, you'll notice the URL to the
+app's service (`/tutorial.PersonSearchService`), followed by the name of the
+invoked RPC (`Search`).
+
+The `content-type`, which is set by the gRPC library, informs Wireshark that the
+HTTP2 message content is a gRPC message. By examining the decoded Protocol
+Buffers message of the sample gRPC request, you can see that the search request
+is for the names "Jason" and "Lily".
+
+### Decoding the server-streamed response
+
+<!--
+TODO: What are the steps that the user needs to follow before the user can see
+      the decoded message as shown in the screenshot? Is the text given below correct?
```

Comment:
After running 'decode as', user can select the packet in the packet list, the detail of the message will show in the detail panel. (Please refer to the *gif* files in my previous reply).
Your text below is OK.